reconcile/utils/jira_client.py

```diff
@@ -22,20 +22,31 @@ def __init__(
         token_auth = self.secret_reader.read(token)
         self.jira = JIRA(self.server, token_auth=token_auth)
 
-    def get_issues(self, fields: Optional[Mapping] = None) -> GottenIssue:
+    def get_issues(self, fields: Optional[Mapping] = None) -> list[Issue]:
         block_size = 100
         block_num = 0
 
-        all_issues: GottenIssue = []
+        all_issues: list[Issue] = []
         jql = "project={}".format(self.project)
         kwargs: dict[str, Any] = {}
         if fields:
             kwargs["fields"] = ",".join(fields)
         while True:
             index = block_num * block_size
-            issues = self.jira.search_issues(jql, index, block_size, **kwargs)
-            # TODO: investigate why the arg-type ignore is needed in CI
-            all_issues.extend(issues)  # type: ignore[arg-type]
+            issues: Union[dict[str, Any], ResultList[Issue]] = self.jira.search_issues(
+                jql, index, block_size, **kwargs
+            )
+            if not isinstance(issues, ResultList):
+                # if search_issues was executed with json_result=True, then we have a Dict.
+                # However, we require a ResultList.
+                # See https://github.com/pycontribs/jira/commit/cc2508485c12232a4a9c4a56ee74175ed818ee20
+                logging.warning(
+                    "Jira client did not receive a ResultList."
+                    " Maybe the call was made with json_result=True which is"
+                    " currently not supported."
+                )
+                continue
```

Comment:
Adding continue here changes the control structure of that loop. Before the if-condition on line 50 was always reached. Could that be an issue?

Reply:
Before the if condition would not have reached if a non-ResultList was returned. The `extend()` call would break on non-iterable dict

Reply:
That makes sense! one last question, what if nothing but ResultLists are returned?

Reply:
> what if nothing but ResultLists are returned?

Do you by chance mean "nothing but Dicts are returned"? Currently we only have ResultLists, because `json_result=True` is nowhere being used. Our logic works for that use-case. Dicts are not supported by us and you would just see a lot of warnings in the logs.

If we want to support `json_result=True`, then we must make adjustments here - it would require us to understand what those Dicts consist of. However, the purpose of this PR was solely to make sure we are working on only objects we expect, without breaking already existing behavior.

Reply:
To be honest, I feel like this is a weakness of the jira module API to return a union like `Union[Dict, List]`. Those types are fundamentally different and should not be returned by the same function. But yeah this is an naive outside view - might be there is a good reason for doing this :)